src/rdkafka_ssl.c

```
@@ -1298,6 +1352,47 @@ int rd_kafka_ssl_ctx_init (rd_kafka_t *rk, char *errstr, size_t errstr_size) {
         }
 #endif
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+        if (rk->rk_conf.ssl.engine_location && !rk->rk_conf.ssl.engine) {
+                char *engine_errstr = NULL;
+
+                /* Openssl loads an engine as dynamic id and stores it in internal
+                 * list, as per LIST_ADD command below. If engine already exists
+                 * in internal list, it is supposed to be fetched using engine id. 
+                 */
+                rk->rk_conf.ssl.engine = ENGINE_by_id(rk->rk_conf.ssl.engine_id);
+                if (!rk->rk_conf.ssl.engine) {
+                        rk->rk_conf.ssl.engine = ENGINE_by_id("dynamic");
+                        if (!rk->rk_conf.ssl.engine)
+                                engine_errstr = "ENGINE_by_id failed";
+                }
+
+                if (!engine_errstr && 
```

Exit out early on failure instead of continuing and checking engine_errstr.